openfisca_core/scripts/__init__.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import importlib
+
+
+TAX_BENEFIT_SYSTEM_OPTIONS = {
+    'country_package': {
+        'short': 'c',
+        'help': 'country package to use. If not provided, an automatic detection will be attempted by scanning the python packages installed in your environment which name contains the word "openfisca".'
+        },
+    'extensions': {
+        'short': 'e',
+        'help': 'extensions to load',
+        'nargs': '*'
+        },
+    'reforms': {
+        'short': 'r',
+        'help': 'reforms to apply to the country package',
+        'nargs': '*'
+        }
+    }
+
+
+def add_tax_benefit_system_arguments(parser):
+    for option, properties in TAX_BENEFIT_SYSTEM_OPTIONS.iteritems():
+        parser.add_argument(
+            '-{}'.format(properties['short']),
+            '--{}'.format(option),
+            action = 'store',
+            help = properties['help'],
+            nargs = properties.get('nargs')
+            )
+
+    return parser
+
+
+def build_tax_benefit_sytem(country_package_name, extensions, reforms):
+    if country_package_name:
+        try:
+            country_package = importlib.import_module(country_package_name)
+            assert hasattr(country_package, 'CountryTaxBenefitSystem')
+        except:
+            print('ERROR: `{}` does not seem to be a valid Openfisca country package.'.format(country_package_name))
+            sys.exit(1)
+    else:
+        country_package_name = detect_country_package()
+        country_package = importlib.import_module(country_package_name)
+
+    tax_benefit_system = country_package.CountryTaxBenefitSystem()
+
+    if extensions:
+        for extension in extensions:
+            tax_benefit_system.load_extension(extension)
+
+    if reforms:
+        for reform_path in reforms:
+            try:
+                [reform_package, reform_name] = reform_path.rsplit('.', 1)
+                reform_module = importlib.import_module(reform_package)
+                reform = getattr(reform_module, reform_name)
+                tax_benefit_system = reform(tax_benefit_system)
+            except:
```

Comment:
The catch-all is very sneaky, could catch any error you don't want to catch, and I would always insist to avoid it!

> Each line can fail, and respectively raise a ValueError, AttributeError, ImportError or TypeError.

```python
invalid_error_message = u'ERROR: ...'.format(...)

tax_benefit_system = None

try:
    [reform_package, reform_name] = reform_path.rsplit('.', 1)
except ValueError:
    log.exception(invalid_reform_message)
else:
    try:
        reform_module = importlib.import_module(reform_package)
    except ImportError:
        log.exception(invalid_reform_message)
    else:
        reform = getattr(reform_module, reform_name, None)
        if reform is not None:
            try:
                tax_benefit_system = reform(tax_benefit_system)
            except Exception:
                log.exception(invalid_reform_message)

return tax_benefit_system
```

This is absolutely not tested and I agree it's longer, but more secure, and can be perhaps simplified.